### PR TITLE
fix for unable to create portable class library projects in VS2015 & Dev15

### PR DESCRIPTION
**Pull request by rohit21agrawal**
Fixes issue : https://github.com/NuGet/Home/issues/3094

PCL projects template creating code calls into VsPackageInstaller.InstallPackagesFromRegistryRepository and passes isUnzipped = true which causes NuGet to assume the v2 version folder format while looking for packages at C:\Program Files (x86)\Microsoft SDKs\NuGetPackages

However, the SDK installs these packages in the v3 version folder format causing NuGet to throw an invalid operation  exception

This fix involves removing the assumption of using the v2 version folder format when isUnzipped=true is passed. Instead it determines the actual feed type using the logic build in the FeedTypeUtility class.

CC: @emgarten @joelverhagen @alpaix @yishaigalatzer @rrelyea

### CODE REVIEW

**Comment by alpaix** [comment]
:shipit:

**Comment by joelverhagen** [comment]
Can you leave a comment about this change in behavior? Also, this still feels like a hack... do we have a work item to fix the installers and revert this change? Or do we intend on this being the functionality Forever?

File: src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs [hunk 1]
```diff
@@ -159,8 +160,17 @@ public void InstallPackagesFromRegistryRepository(string keyName, bool isPreUnzi
                 throw new ArgumentException(CommonResources.Argument_Cannot_Be_Null_Or_Empty, nameof(packageVersions));
             }
 
-            PumpingJTF.Run(() =>
+            PumpingJTF.Run(async () =>
                 {
+                    // HACK !!! : This is a hack for PCL projects which send isPreUnzipped = true, but their package source 
+                    // (located at C:\Program Files (x86)\Microsoft SDKs\NuGetPackages) follows the V3
+                    // folder version format.
+                    var isProjectJsonProject = await EnvDTEProjectUtility.HasBuildIntegratedConfig(project);
```
Comment (emgarten) [inline src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs]: You only need this check if isPreUnzipped is true

**Comment by emgarten** [comment]
Optimize the check, then :shipit:

**Comment by rohit21agrawal** [comment]
@joelverhagen discussed this with @emgarten , it seems that there has been no change in behavior in the installer or the API calls . Since,  Dev14 has already shipped , we will have to keep this hack forever in our code :( .

**Comment by joelverhagen** [comment]
thanks for the clarification @rohit21agrawal, LGTM

**Pull request merged**